feat: Add better docs for setting the user (#12224)
getsentry/sentry-docs: docs/platforms/javascript/common/enriching-events/identify-user/index.mdx

If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server.

If the field is omitted, the default value is `"{{auto}}"`. SDKs and other clients should not rely on this behavior and should set IP addresses or `"{{auto}}"` explicitly.

> To ensure your users' IP addresses are never stored in your event data, you can go to your project settings, click on "Security & Privacy", and enable "Prevent Storing of IP Addresses" or use Sentry's [server-side data](/security-legal-pii/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.

Reviewer: ```suggestion
To ensure your users' IP addresses are never stored in your event data, you can go to your project settings, click on "Security & Privacy", and enable "Prevent Storing of IP Addresses" or use Sentry's [server-side data scrubbing](/security-legal-pii/scrubbing/) to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.
```

</DefinitionList>

Additionally, you can provide arbitrary key/value pairs beyond the reserved names, and the Sentry SDK will store those with the user.

## Setting the User

You can set the user by calling the `setUser` method on the Sentry SDK:

<PlatformContent includePath="enriching-events/set-user" />

You can also clear the currently set user:

<PlatformContent includePath="enriching-events/unset-user" />

<PlatformCategorySection supported={["server"]}>

> `Sentry.setUser()` will set the user for the currently active request - see <PlatformLink to="/enriching-events/request-isolation">Request Isolation</PlatformLink> for more information. For example, if you want to set the user for a single request, you can do this like this:

Reviewer: Are there any additional tasks required for non-request tasks (like async handlers or whatever) that we need to mention?

Author: I would say just the stuff that is generally written on the request isolation page, that should cover that hopefully?

<PlatformContent includePath="enriching-events/set-user-request" />

> Or if you want to set the user for all requests, you could do this with a middleware like this:

Reviewer 2: ```suggestion
Or if you want to set the user for all requests, you could use a middleware like this:
```
